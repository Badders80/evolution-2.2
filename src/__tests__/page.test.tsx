// Mock Next.js Image component
/* eslint-disable @next/next/no-img-element */
jest.mock("next/image", () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    fill,
    ...props
  }: {
    src: string;
    alt: string;
    fill?: boolean;
    [key: string]: unknown;
  }) => <img src={src} alt={alt} {...(fill ? { style: { objectFit: "cover" } } : {})} {...props} />,
}));
/* eslint-enable @next/next/no-img-element */

import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import HomePage from "../app/page";

describe("HomePage", () => {
  it("renders the header with logo and navigation", () => {
    render(<HomePage />);

    // Check logo
    const logo = screen.getByAltText("Evolution Stables Logo");
    expect(logo).toBeInTheDocument();

    // Check nav links - be more specific
    expect(screen.getByRole("link", { name: "Services" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Process" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Pricing" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Blog" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Contact" })).toBeInTheDocument();

    // Check CTA button
    expect(screen.getByText("Get Started")).toBeInTheDocument();
  });

  it("renders the hero section with brand logo and tagline", () => {
    render(<HomePage />);

    // Check for the brand logo alt text
    expect(screen.getByAltText("Evolution Stables Brand Logo")).toBeInTheDocument();
    // Check that the TextGenerateEffect component is present (it may not show text immediately in tests)
    expect(screen.getByText("|")).toBeInTheDocument();
  });

  it("renders section placeholders", () => {
    render(<HomePage />);

    // Check section headings (be more specific to avoid conflicts with nav links)
    expect(screen.getByRole("heading", { name: "Services", level: 2 })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: "Process", level: 2 })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: "Pricing", level: 2 })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: "Blog", level: 2 })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: "Contact", level: 2 })).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<HomePage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
