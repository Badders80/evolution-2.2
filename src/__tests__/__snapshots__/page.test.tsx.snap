// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomePage matches snapshot 1`] = `
<div
  class="min-h-screen flex flex-col"
>
  <header
    class="w-full bg-gray-200 flex items-center justify-between px-6 py-4"
  >
    <div
      class="flex items-center"
    >
      <img
        alt="Evolution Stables Logo"
        height="40"
        src="/assets/Logo-Black.png"
        width="40"
      />
    </div>
    <nav
      class="text-center flex-1"
    >
      <span
        class="text-black font-medium"
      >
        Nav Bar
      </span>
    </nav>
    <div
      class="text-sm text-gray-700"
    >
      Get Started / Login
    </div>
  </header>
  <section
    class="relative w-full flex items-center justify-center h-screen bg-black text-gold"
    style="background-image: url("/assets/Horse-Double-Black.png"); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <h1
      class="text-3xl font-bold"
    >
      Hero
    </h1>
  </section>
  <section
    class="w-full py-section bg-gray-500 text-gold text-lg"
  >
    <div
      class="container"
    >
      Section1
    </div>
  </section>
  <section
    class="w-full py-section bg-gray-700 text-gold text-lg"
  >
    <div
      class="container"
    >
      Section2
    </div>
  </section>
  <section
    class="w-full py-section bg-gray-500 text-gold text-lg"
  >
    <div
      class="container"
    >
      Section3
    </div>
  </section>
  <section
    class="w-full py-section bg-gray-700 text-gold text-lg"
  >
    <div
      class="container"
    >
      Section4
    </div>
  </section>
  <section
    class="w-full py-section bg-gray-500 text-gold text-lg"
  >
    <div
      class="container"
    >
      CTA
    </div>
  </section>
  <footer
    class="w-full py-section bg-gray-700 text-gold text-lg text-center"
  >
    Footer
  </footer>
</div>
`;
